File: exercism/c/darts/src/darts.c
#include "darts.h"
#include <math.h>

int score(coordinate_t position) {
  float radius = sqrt(powf(position.x, 2) + powf(position.y, 2));

  if (radius <= 10.0 && radius > 5.0)
    return 1;
  else if (radius <= 5.0 && radius > 1.0)
    return 5;
  else if (radius <= 1.0 && radius >= 0.0)
    return 10;
  else
    return 0;
}

File: exercism/go/darts/darts.go
package darts

import "math"

// Score gives the dart score for the given coordinates
func Score(x, y float64) int {
	distance := math.Sqrt(math.Pow(x, 2) + math.Pow(y, 2))

	switch {
	case distance <= 1:
		return 10
	case distance <= 5:
		return 5
	case distance <= 10:
		return 1
	default:
		return 0
	}
}
